frontend/src/metabase/lib/expressions/compile.js

```diff
@@ -89,26 +90,7 @@ class ExpressionMBQLCompilerVisitor extends ExpressionCstVisitor {
 
   dimensionExpression(ctx) {
     const name = this.visit(ctx.dimensionName);
-    if (ctx.resolveAs === "metric") {
-      const metric = parseMetric(name, this._options);
-      if (!metric) {
-        throw new Error(`Unknown Metric: ${name}`);
-      }
-      return ["metric", metric.id];
-    } else if (ctx.resolveAs === "segment") {
-      const segment = parseSegment(name, this._options);
-      if (!segment) {
-        throw new Error(`Unknown Segment: ${name}`);
-      }
-      return ["segment", segment.id];
-    } else {
-      // fallback
-      const dimension = parseDimension(name, this._options);
-      if (!dimension) {
-        throw new Error(`Unknown Field: ${name}`);
-      }
-      return dimension.mbql();
-    }
+    return this.resolve(ctx.resolveAs, name);
```

Comment:
We're one step closer to eliminate `dimensionExpression` alltogether (as [I mentioned earlier](https://github.com/metabase/metabase/pull/14232#discussion_r551505845)) and hopefully the parser/type-checker/compiler can work directly with `identity` literal.